src/js/jsx/shared/Dialog.jsx

```diff
@@ -211,34 +211,40 @@ define(function (require, exports, module) {
-                        dialogComputedStyle = window.getComputedStyle(dialogEl),
-                        dialogMarginTop = math.pixelDimensionToNumber(dialogComputedStyle.marginTop),
-                        dialogMarginBottom = math.pixelDimensionToNumber(dialogComputedStyle.marginBottom),
-                     
+                    var clientHeight = window.document.documentElement.clientHeight,
                         // Adjust the position of the opened dialog according to the target
                         targetBounds = this._target.getBoundingClientRect(),
                         offsetParentBounds = this._target.offsetParent.getBoundingClientRect(),
                         placedDialogTop = Math.round(targetBounds.bottom) - Math.round(offsetParentBounds.top),
-                        placedDialogBottom = placedDialogTop + Math.round(dialogBounds.height);
-
-                    if (placedDialogBottom > clientHeight) {
-                        // If there is space, let's place this above the target
+                        placedDialogBottom = placedDialogTop + Math.round(dialogBounds.height),
+                        topPosition,
+                        bottomPosition,
+                        placedAbove;
+                    
+                    // If there is space below the target, place the dialog below.
+                    if (placedDialogBottom <= clientHeight) {
+                        placedAbove = false;
+                        topPosition = placedDialogTop;
+                    } else {
                         placedAbove = true;
+                        
+                        // Need to account for element margin
+                        var dialogComputedStyle = window.getComputedStyle(dialogEl),
+                            dialogMarginTop = math.pixelDimensionToNumber(dialogComputedStyle.marginTop),
+                            dialogMarginBottom = math.pixelDimensionToNumber(dialogComputedStyle.marginBottom);
+
+                        // If there is space above the target, let's place the dialog above it
                         if (dialogBounds.height + dialogMarginTop + dialogMarginBottom < targetBounds.top) {
-                            placedDialogTop = targetBounds.top -
-                                dialogBounds.height - dialogMarginTop - dialogMarginBottom;
+                            bottomPosition = clientHeight - targetBounds.top - dialogMarginBottom ;
                         } else {
-                            placedDialogTop = clientHeight -
+                            topPosition = clientHeight -
                                 dialogBounds.height - dialogMarginTop - dialogMarginBottom;
                         }
                     }
 
                     this.setState({
-                        "topPosition": placedDialogTop + "px",
+                        "topPosition": topPosition + "px",
+                        "bottomPosition": bottomPosition + "px",
```

Comment:
I can't find any user-facing issue from this... but it is possible that `topPosition` and `bottomPosition` to be accessed without being defined... resulting in the string "undefinedpx" being added to state.
![ss](http://cl.ly/271X1P2F3H1G/Image%202015-12-09%20at%203.56.19%20PM.png)